### PR TITLE
minor bug fixes

### PR DESCRIPTION
**Pull request by seb-wilkes**

### CODE REVIEW

**Comment by seb-wilkes** [comment]
btw can give you me write access please? 

also my supervisor and I last year had a great rule where we only accepted the other person's PR

**Comment by seb-wilkes** [comment]
**wait a mo, let me confirm something first**

**Comment by seb-wilkes** [comment]
okay follow up PR has been tested, this is good to go

**Comment by vdlad** [comment]
oh ofc - sorry i didnt realize this wasn't the repo that was shared

File: whitebox_inversion/utils.py [hunk 1]
```diff
@@ -1,36 +1,42 @@
...
+    one_hot = torch.full((batch_size, seq_len, vocab_size),
+                         fill_value=cold_val,
+                         device=device, dtype=torch.float32)    
+    # Use scatter to set the 'hot' values, factoring in normalisation condition
+    one_hot.scatter_(2, tokens.unsqueeze(-1), cold_val + (1.0-cold_val)*hot_val)    
+    return one_hot
 
 def simplex_projection(tensor: torch.Tensor) -> torch.Tensor:
-    d_batch, d_tokens, d_vocab = tensor.shape
+    batch, sequence, dims = tensor.shape
```
Comment (vdlad) [inline whitebox_inversion/utils.py:17]: I feel like i prefer the d_batch etc cause it dimensionality of the stuff etc - also dims is a bit charged cause "dim" is a flag - also dim in this case is the d_vocab

File: whitebox_inversion/utils.py [hunk 2]
```diff
@@ -1,36 +1,42 @@
...
     needs_projection = (norm_s_c < R).float()
     does_not_need_projection = 1 - needs_projection
-    scaled_s = torch.where(needs_projection.bool(), (R / norm_s_c) * (tensor - c) + c, tensor)
+    scaled_s = torch.where(needs_projection.bool(), (R / norm_s_c) * (s - c) + c, s)
     projection = simplex_projection(scaled_s)
-    result = does_not_need_projection * tensor + needs_projection * projection
-    return result
\ No newline at end of file
+    result = does_not_need_projection * s + needs_projection * projection
+    return result.view(original_shape)
```
Comment (vdlad) [inline whitebox_inversion/utils.py:41]: why are you adding a view here? also what does the variable s mean?

File: whitebox_inversion/utils.py [hunk 3]
```diff
@@ -1,36 +1,42 @@
...
+    rho = torch.clamp((mu > threshold).sum(dim=-1) - 1, 0, dims-1)
     threshold_per_row = torch.gather(threshold, 2, rho.unsqueeze(-1))
-    projected = torch.max(tensor - threshold_per_row, torch.zeros_like(tensor))
-    return projected
+    return torch.maximum(tensor - threshold_per_row, torch.zeros(1, dtype=tensor.dtype, device=tensor.device))
 
 def entropy_projection(tensor: torch.Tensor, entropy: float) -> torch.Tensor:
-    positive_mask = (tensor > 0).float()
+    original_shape = tensor.shape
+    s = tensor.view(original_shape[0], -1, original_shape[-1])
```
Comment (vdlad) [inline whitebox_inversion/utils.py:28]: porque?

Comment (vdlad) [reply whitebox_inversion/utils.py:28]: I dont think it needs a view - its already just dim = 3

File: whitebox_inversion/utils.py (hunk 1 above)
Comment (seb-wilkes) [reply whitebox_inversion/utils.py:17]: oh yeah sorry this was just a copy-paste mistake
